app/repositories/ArrivalMovementRepository.scala

```diff
@@ -315,210 +312,135 @@ class ArrivalMovementRepository @Inject()(
+          Updates.inc("nextMessageCorrelationId", 1),
+          Updates.push("messages", message)
         )
       )
+      .toFuture()
+      .map {
+        result =>
+          if (result.wasAcknowledged()) {
+            if (result.getModifiedCount == 0) Failure(new Exception(s"Could not find arrival $arrivalId"))
+            else Success(())
+          } else Failure(new Exception("Failed to update arrival"))
 
-    collection.flatMap {
-      _.simpleFindAndUpdate(
-        selector = selector,
-        update = modifier
-      ).map {
-        _.lastError
-          .map {
-            le =>
-              if (le.updatedExisting) Success(()) else Failure(new Exception(s"Could not find arrival $arrivalId"))
-          }
-          .getOrElse(Failure(new Exception("Failed to update arrival")))
       }
-    }
-  }
 
   def addResponseMessage(arrivalId: ArrivalId, message: MovementMessage): Future[Try[Unit]] = {
-    val selector = Json.obj(
-      "_id" -> arrivalId
+    val selector = Filters.eq("_id", arrivalId)
+    val modifier = Updates.combine(
+      Updates.set("updated", message.received.get),
+      Updates.set("lastUpdated", message.received.get),
+      Updates.push("messages", message)
     )
+    collection
+      .updateOne(filter = selector, update = modifier)
+      .toFuture()
+      .map {
+        result =>
+          if (result.wasAcknowledged()) {
+            if (result.getModifiedCount == 0) Failure(new Exception(s"Could not find arrival $arrivalId"))
+            else Success(())
+          } else Failure(new Exception("Failed to update arrival"))
 
-    val modifier =
-      Json.obj(
-        "$set" -> Json.obj(
-          "updated"     -> message.received.get,
-          "lastUpdated" -> message.received.get
-        ),
-        "$push" -> Json.obj(
-          "messages" -> Json.toJson(message)
-        )
-      )
-
-    collection.flatMap {
-      _.simpleFindAndUpdate(
-        selector = selector,
-        update = modifier
-      ).map {
-        _.lastError
-          .map {
-            le =>
-              if (le.updatedExisting) Success(()) else Failure(new Exception(s"Could not find arrival $arrivalId"))
-          }
-          .getOrElse(Failure(new Exception("Failed to update arrival")))
       }
-    }
   }
 
   def arrivalsWithoutJsonMessagesSource(limit: Int): Future[Source[Arrival, Future[Done]]] = {
 
-    val messagesWithNoJson =
-      Json.obj(
-        "messages" -> Json.obj(
-          "$elemMatch" -> Json.obj(
-            "messageJson" -> Json.obj(
-              "$exists" -> false
-            )
-          )
-        )
-      )
-
-    val messagesWithEmptyJson =
-      Json.obj(
-        "messages" -> Json.obj(
-          "$elemMatch" -> Json.obj(
-            "messageJson" -> Json.obj()
-          )
-        )
-      )
-
-    val query = Json.obj("$or" -> Json.arr(messagesWithNoJson, messagesWithEmptyJson))
+    // TODO: is this correct?
```

Comment:
Does this need looking or is it just a redundant comment?

Reply:
Not sure if we need this anymore - need to speak to frontend to see if they call it.